torch_xla/csrc/tensor.cpp

```diff
@@ -1649,12 +1669,13 @@ std::shared_ptr<XLATensor::Async> XLATensor::SyncTensorsGraphInternal(
       "SyncTensorsGraphInternal", tensorflow::profiler::TraceMeLevel::kInfo);
   SyncTensorCollection coll = CollectSyncTensors(*tensors, config);
   if (coll.indices.empty()) {
+    TensorCollectionBarrier(&coll);
```

Comment:
Do we know why the barrier here is necessary?

Reply:
Please add a comment here explaining why it is necessary, otherwise lgtm.

Reply:
Context: 
This check is necessary to ensure previous execution completes before exiting `SyncTensorsGraphInternal`. 

Specifically, this check prevents a corner case scenario that appeared in the eager mode tests in which  `get_handle()` causes a seg fault via the following call sequence: `GetTensors` -> `GetTensorsFused` -> `SyncTensorsGraphInternal` then `xla::ComputationClient::Get()->TransferFromServer()`. Although the issue is captured in our eager mode tests, it can happen in the default execution mode too.